client/my-sites/marketing/connections/service.jsx

```diff
@@ -276,7 +279,13 @@ export class SharingService extends Component {
 				requestExternalAccess( connection.refresh_URL, () => {
```

Comment:
`this.setState( { isRefreshing: true } );` should be moved inside the `requestExternalAccess` function. Otherwise a missing keyring connection ID can get the button stuck in a disabled state.

Reply:
Good catch!